multicluster/controllers/multicluster/internal/remote_cluster_manager.go

```diff
@@ -0,0 +1,154 @@
+package internal
+
+import (
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"context"
+	"github.com/go-logr/logr"
+	"sync"
+)
+
+type clusterEvent struct {
+	isAdd         bool
+	remoteCluster RemoteCluster
+}
+
+type RemoteClusterManager interface {
+	// Start RemoteClusterManager on an event loop in a go routine
+	Start() error
+	// Stop RemoteClusterManager
+	Stop() error
+	// AddRemoteCluster adds a member cluster to RemoteClusterManager.
+	AddRemoteCluster(remoteCluster RemoteCluster)
+	// RemoveRemoteCLuster removes a remote cluster from RemoteClusterManager.
+	RemoveRemoteCluster(remoteCluster RemoteCluster)
+	// GetRemteClusters returns all remote clusters
+	GetRemoteClusters() map[common.ClusterID]RemoteCluster
+	// GetElectedLeaderClusterID returns the leader remote cluster or INVALID_CLUSTER_ID if none elected
+	GetElectedLeaderClusterID() common.ClusterID
+	// GetLocalClusterID returns local cluster ID
+	GetLocalClusterID() common.ClusterID
+}
+
+// The manager manages all leaders defined in the cluster set and manages common area access into them
```

Comment:
Nit - better to add "." at the end of sentences, esp. when your comments have multiple sentences.